### ISSUE TITLE
Build on 1.30.0 fails: `ares_event_configchg.c:370: error: ‘F_GETFL’ undeclared (first use in this function)`

### ISSUE THREAD

**Issue by barracuda156** [c-ares/c-ares#784]
```
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_dns_write.c: In function ‘ares_dns_write_rr’:
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_dns_write.c:939: warning: passing argument 2 of ‘ares__buf_append_be16’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_dns_write.c:946: warning: passing argument 2 of ‘ares__buf_append_be16’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_dns_write.c:965: warning: passing argument 2 of ‘ares__buf_append_be16’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_dns_write.c:1048: warning: passing argument 2 of ‘ares__buf_append_be16’ with different width due to prototype
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c: In function ‘ares_event_configchg_init’:
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:370: warning: implicit declaration of function ‘fcntl’
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:370: warning: nested extern declaration of ‘fcntl’
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:370: error: ‘F_GETFL’ undeclared (first use in this function)
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:370: error: (Each undeclared identifier is reported only once
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:370: error: for each function it appears in.)
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:371: error: ‘F_SETFL’ undeclared (first use in this function)
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_net_c-ares/c-ares/work/c-ares-1.30.0/src/lib/ares_event_configchg.c:371: error: ‘O_NONBLOCK’ undeclared (first use in this function)
:info:build make[2]: *** [src/lib/CMakeFiles/c-ares.dir/ares_event_configchg.c.o] Error 1
:info:build make[2]: *** Waiting for unfinished jobs....
```

**Comment by barracuda156** [comment]
Possibly same as https://github.com/c-ares/c-ares/issues/782

**Comment by bradh352** [comment]
yeah, duplicate of #782
